src/Compilers/CSharp/Portable/Emitter/Model/AttributeDataAdapter.cs

```diff
@@ -28,8 +28,22 @@ ImmutableArray<Cci.IMetadataExpression> Cci.ICustomAttribute.GetArguments(EmitCo
             return builder.ToImmutableAndFree();
         }
 
-        Cci.IMethodReference Cci.ICustomAttribute.Constructor(EmitContext context)
+        Cci.IMethodReference Cci.ICustomAttribute.Constructor(EmitContext context, bool reportDiagnostics)
         {
+            if (this.AttributeConstructor.IsDefaultValueTypeConstructor())
+            {
+                // Parameter constructors for structs exist in symbol table, but are not emitted.
+                // Produce an error since we cannot use it (instead of crashing):
+                // Details: https://github.com/dotnet/roslyn/issues/19394
+
+                if (reportDiagnostics)
+                {
+                    context.Diagnostics.Add(ErrorCode.ERR_NotAnAttributeClass, NoLocation.Singleton, this.AttributeClass);
+                }
+
+                return null;
+            }
+
             PEModuleBuilder moduleBeingBuilt = (PEModuleBuilder)context.Module;
             return (Cci.IMethodReference)moduleBeingBuilt.Translate(this.AttributeConstructor, (CSharpSyntaxNode)context.SyntaxNodeOpt, context.Diagnostics);
```

Comment:
>Translate [](start = 58, length = 9)

Please add an assert into this method that it is not called with DefaultValueTypeConstructor